test/integration/oidc/test_auth_oidc.py

```diff
@@ -100,6 +102,12 @@ def setUpClass(cls):
             cls.disableOauthlibHttps()
             cls.container_name = f"{cls.__name__}_container"
             start_keycloak_docker(container_name=cls.container_name)
+
+            # Dynamically find a free port
+            cls.random_port = cls._find_free_port()
```

Comment:
This is a great improvement! We should be able to get rid of the unnecessary restart as well? We no longer need `configure_oidc_and_restart` right?

Reply:
I checked but I think the restart is still needed to get the app to load the generated OIDC config files